### PR TITLE
Bug 1909027: Modifying selected disk count based on selected disk type

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1909027

### CODE REVIEW

**Review by afreen23 — APPROVED** [review]
Can you make it a single commit ?
The changes looks good.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx [hunk 1]
```diff
@@ -40,7 +41,14 @@ export const DiscoveryDonutChart: React.FC<DiscoveryDonutChartProps> = ({ state,
         if (nodes.includes(disk.node)) {
           const isValidSize =
             Number(disk.size) >= minSize && (maxSize ? Number(disk.size) <= maxSize : true);
-          if (isValidSize) {
+
+            // For disk type All it always true
+            let diskTypeMatched: boolean = true;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx:46]:
```suggestion
            let hasDiskType: boolean = true;
```

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx [hunk 2]
```diff
@@ -40,7 +41,14 @@ export const DiscoveryDonutChart: React.FC<DiscoveryDonutChartProps> = ({ state,
...
+            if (DISK_TYPES[state.diskType]?.property) {
+              diskTypeMatched = DISK_TYPES[state.diskType].property === disk.property ? true : false;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx:48]:
```suggestion
              hasDiskType = DISK_TYPES[state.diskType].property === disk.property;
```

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx (hunk 1 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx:46]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx:48]: ack

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx:48]: My bad, I dont know why i put true and false here :)

**Comment by GowthamShanmugam** [comment]
@afreen23 changes are done

**Pull request merged**